cf/data/data.py

```diff
@@ -1374,41 +1374,58 @@ def _del_dask(self, default=ValueError(), conform=True):
     def _del_cached_elements(self):
         """Delete any cached element values.
 
-        Updates *data* in-place to remove the cached element values
-        ``'first_element'``, ``'second_element'`` and
-        ``'last_element'``.
-
-        .. note:: By default, `_del_cached_elements` is run whenever
-                  the `_set_dask` and `del_dask` methods are used. If
-                  the `dask` array is updated or changed without using
-                  the default behaviour of either of these two
-                  methods, and there is any chance that the cached
-                  values might be inconsistent with the new data, then
-                  `_del_cached_elements` must be called explicitly to
-                  ensure consistency.
+        Updates *data* in-place to remove the cached element values.
 
         .. versionadded:: 3.14.0
 
-        .. seealso:: `_del_dask`, `_set_cached_elements`, `_set_dask`
+        .. seealso:: `_get_cached_elements`, `_set_cached_elements`
 
         :Returns:
 
             `None`
 
         """
-        custom = self._custom
-        for element in ("first_element", "second_element", "last_element"):
-            custom.pop(element, None)
+        self._custom.pop("cached_elements", None)
+
+    def _get_cached_elements(self):
+        """Return the cache of selected element values.
+
+        .. versionadded:: 3.14.1
+
+        .. seealso:: `_del_cached_elements`, `_set_cached_elements`
+
+        :Returns:
+
+            `dict`
+                The cached element values
+
+        **Examples**
+
+        >>> d._get_cached_elements()
+        {}
+
+        >>> d._get_cached_elements()
+        {0: 273.15, 1: 274.56, -1: 269.95}
```

Comment:
Can users specify to cache values that aren't the first, second or last somehow (or will we ever open up that possibility)? Because then the negative indices could lead to trouble, for example say the array is size `n` then the equivalent index `n-1` would be stored separately in the dictionary and the two could get out of sync. So possibly we'd need some extra logic to handle that, in which case some updates along those lines might be required?

Reply:
They can indeed, but I'm not worried about them getting out of sync because we already do this for size 1 arrays (i.e. we store `0` and `-1` separately with the same value; and also all of this is "protected" by underscores, so buyer beware! How does that sound?

Reply:
Aha, good point (and good illustration RE the size-1 array)! Internals should not be messed with and if the user does this it's their own fault if something goes wrong, I can't argue with that :smile: That's fine by me, then. Thanks for clarifying. Marking as resolved.